lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -186,10 +205,13 @@ def __init__(
         """
         super().__init__(charm, "kubernetes-service-patch")
         self.charm = charm
-        self.service_name = service_name if service_name else self._app
+        self.service_name = service_name or self._app
+        if self.service_name == self._app and service_type == "LoadBalancer":
```

Comment:
Add comment? E.g.
```suggestion
        # The LB service cannot have the same name as the app, because...
        if self.service_name == self._app and service_type == "LoadBalancer":
```